public/pages/CreatePolicy/containers/CreatePolicy/CreatePolicy.tsx

```diff
@@ -78,6 +80,22 @@ export default class CreatePolicy extends Component<CreatePolicyProps, CreatePol
     this._isMount = false;
   }
 
+  componentDidUpdate(prevProps: CreatePolicyProps, prevState: Readonly<CreatePolicyState>) {
+    if (prevProps.dataSourceId != this.props.dataSourceId) {
+      // reset the state, if dataSourceId changes, i.e., clear state
+      this.setState({
```

Comment:
can we have a common function to set default state, as we are setting state in constructor and here ? I think this pattern is repeated across components.

Reply:
I'm not resetting the complete state but I've abstracted out the base state separately to reduce duplication